pg_metadata/dock.py

```diff
@@ -314,3 +316,39 @@ def set_html_content(self, title=None, body=None):
     def default_html_content(self):
         self.set_html_content(
             'PgMetadata', tr('You should click on a layer in the legend which is stored in PostgreSQL.'))
+
+    @staticmethod
+    def set_layer_metadata(
+            layer: QgsVectorLayer,
+            connection: QgsAbstractDatabaseProviderConnection,
+            schema: str,
+            table: str,
+    ):
+        """ Set the main owner contact for the attribution. """
+        sql = (
+            "SELECT name FROM pgmetadata.v_contact "
+            f"WHERE table_name = '{table}' "
+            f"AND schema_name = '{schema}' "
+            "ORDER BY CASE "
+            "WHEN contact_role_code = 'OW' then 1 "
+            "WHEN contact_role_code = 'DI' then 2 "
+            "WHEN contact_role_code = 'CU' then 3 "
+            "ELSE 10 "
+            "END ASC "
+            "LIMIT 1;"
+        )
+
+        data = connection.executeSql(sql)
+        if not data:
+            return
+
+        if data[0] == NULL or data[0][0] == NULL:
+            return
+
+        # QGIS Server panel
+        layer.setAttribution(data[0][0])
+
+        # Metadata panel
+        metadata = layer.metadata()
+        metadata.setRights([data[0][0]])
```

Comment:
Should we always override existing metadata, or only if a new checkbox is checked in options ?

Reply:
Did you successfully run the code above?